fineract-core/src/main/java/org/apache/fineract/infrastructure/event/external/repository/ExternalEventRepository.java

```diff
@@ -40,6 +41,7 @@ public interface ExternalEventRepository extends JpaRepository<ExternalEvent, Lo
     void deleteOlderEventsWithSentStatus(@Param("status") ExternalEventStatus status,
             @Param("dateForPurgeCriteria") LocalDate dateForPurgeCriteria);
 
+    @Transactional
```

Comment:
Kind of useless, no?

Reply:
In the new thread it needs a transaction.

Reply:
frankly speaking, defining the transactional boundary on the repository level is often a code-smell. With transactionality you want to have atomicity across an operation, not a single repository call. I'd move this one level up (and since you're using a lambda there, I'd use TransactionTemplate for it)

Reply:
hmm, yea you're right, thanks for the tip with TrancactionTemplate